anomalib/data/custom.py

```diff
@@ -0,0 +1,490 @@
+            normal (str, optional): Name of the directory containing normal images.
+                Defaults to "normal".
+            abnormal (str, optional): Name of the directory containing abnormal images.
+                Defaults to "abnormal".
+            task (str, optional): Task type. Could be either classification or segmentation.
+                Defaults to "classification".
+            mask_dir (Optional[Union[str, Path]], optional): Path to the directory containing
+                the mask annotations. Defaults to None.
+            extensions (Optional[Tuple[str, ...]], optional): Type of the image extensions to read from the
+                directory. Defaults to None.
+            split_ratio (float, optional): Ratio to split normal training images and add to the
+                test set in case test set doesn't contain any normal images.
+                Defaults to 0.2.
+            seed (int, optional): Random seed to ensure reproducibility when splitting. Defaults to 0.
+            image_size (Optional[Union[int, Tuple[int, int]]], optional): Size of the input image.
+                Defaults to None.
+            train_batch_size (int, optional): Training batch size. Defaults to 32.
+            test_batch_size (int, optional): Test batch size. Defaults to 32.
+            num_workers (int, optional): Number of workers. Defaults to 8.
+            transform_config (Optional[Union[str, A.Compose]], optional): Config for pre-processing.
+                Defaults to None.
+            create_validation_set (bool, optional):Boolean to create a validation set from the test set.
+                MVTec dataset does not contain a validation set. Those wanting to create a validation set
+                could set this flag to ``True``.
+
+        Examples:
+            Assume that we use Custom Dataset for the MVTec/bottle/broken_large category. We would do:
+            >>> from anomalib.data import CustomDataModule
+            >>> datamodule = CustomDataModule(
+            ...     root="./datasets/MVTec/bottle/test",
+            ...     normal="good",
+            ...     abnormal="broken_large",
+            ...     image_size=256
+            ... )
+            >>> datamodule.setup()
+            >>> i, data = next(enumerate(datamodule.train_dataloader()))
+            >>> data["image"].shape
+            torch.Size([16, 3, 256, 256])
+
+            >>> i, test_data = next(enumerate(datamodule.test_dataloader()))
+            >>> test_data.keys()
+            dict_keys(['image'])
+
+            We could also create a Custom DataModule for datasets containing mask annotations.
+            The dataset expects that mask annotation filenames must be same as the original filename.
+            To show an example, we therefore need to modify the mask filenames in MVTec dataset.
+
+            >>> # Rename MVTec mask annotations so that they are the same as image filanames
```

Comment:
I'm afraid the example in the docstring might cause some confusion with the users (why use the custom dataset class for MVTec if there is a dataset class specific for mvtec). Maybe we could keep it simple and start the example with the assumption that the user has a folder of normal images and a folder of abnormal images, and explicitly state this at the beginning of the example.